core/src/main/java/com/sequenceiq/cloudbreak/service/decorator/StackDecorator.java

```diff
@@ -64,6 +69,15 @@ public Stack decorate(Stack subject, Object... data) {
             if (subject.getFailurePolicy() != null) {
                 validatFailurePolicy(subject, subject.getFailurePolicy());
             }
+
+            String rawPassword = String.valueOf(System.currentTimeMillis());
+            try {
+                MessageDigest messageDigest = MessageDigest.getInstance("MD5");
+                byte[] digest = messageDigest.digest(rawPassword.getBytes());
+                subject.setSaltPassword(new BigInteger(1, digest).toString(RADIX));
+            } catch (NoSuchAlgorithmException e) {
+                subject.setSaltPassword(rawPassword);
```

Comment:
In case of no such algorithm (which should never occur, except if some terrible thing happened) we should throw an exception instead of going with a weaker password.